packages/itwin/tree-widget/src/components/trees/stateless/models-tree/ModelsTree.tsx

```diff
@@ -49,18 +55,28 @@ export function StatelessModelsTree({
   filter,
   density,
   hierarchyLevelConfig,
+  hierarchyConfig,
   selectionMode,
   onPerformanceMeasured,
   onFeatureUsed,
 }: StatelessModelsTreeProps) {
-  const { getModelsTreeIdsCache, visibilityHandlerFactory } = useCachedVisibility(activeView);
+  const hierarchyConfiguration = useMemo<ModelsTreeHierarchyConfiguration>(
+    () => ({
+      elementClassGrouping: "enable",
+      elementClassSpecification: "BisCore.GeometricElement3d",
+      showEmptyModels: false,
+      ...hierarchyConfig,
+    }),
+    [hierarchyConfig],
+  );
```

Comment:
Doesn't this mean that consumer will need to pass a stable `hierarchyConfig` object? E.g. this won't work:

```tsx
<StatelessModelsTree hierarchyConfig={{ elementClassGrouping: "disable" }} />
```

Reply:
Switched to comparing values rather than the object.